lib/web_scraper.dart

```diff
@@ -1,167 +1,220 @@
+    assert(_response != null);
+
+    // Quering the list of elements by tag names.
+    var scripts = document.getElementsByTagName('script');
+    var result = <String>[];
+
+    // Looping in all script tags of the document.
+    for (var script in scripts) {
+      /// Adds the data enclosed in script tags
+      /// ex. if document contains <script> var a = 3; </script>
+      /// var a = 3; will be added to result.
+      result.add(script.text);
+    }
+    return result;
+  }
+
+  /// Returns Map between given variable names and list of their occurence in the script tags
+  /// ex. if document contains
+  /// <script> var a = 15; var b = 10; </script>
+  /// <script> var a = 9; </script>
+  /// method will return {a: ['var a = 15;', 'var a = 9;'], b: ['var b = 10;'] }.
+  Map<String, dynamic> getScriptVariables(List<String> variableNames) {
+    // The _response should not be null (loadWebPage must be called before getScriptVariables).
+    assert(_response != null);
+
+    // Quering the list of elements by tag names.
+    var scripts = document.getElementsByTagName('script');
+
+    var result = <String, List<String>>{};
+
+    // Looping in all the script tags of the document.
+    for (var script in scripts) {
+      // Looping in all the variable names that are required to extract.
+      for (var variableName in variableNames) {
+        // regular expression to get the variable names
+        var re = RegExp(
+            '$variableName *=.*?;(?=([^\"\']*\"[^\"\']*\")*[^\"\']*\$)',
+            multiLine: true);
+        //  Iterate all matches
+        Iterable matches = re.allMatches(script.text);
+        matches.forEach((match) {
+          if (match != null) {
+            // list for all the occurence of the variable name
+            var temp = result[variableName];
+            if (result[variableName] == null) {
+              temp = [];
+            }
+            temp.add(script.text.substring(match.start, match.end));
+            result[variableName] = temp;
+          }
+        });
+      }
+    }
+
+    // Returning final result i.e. Map of variable names with the list of their occurences.
+    return result;
+  }
+
+  /// Returns webpage's html in string format.
+  String getPageContent() => _response != null
+      ? _response.body.toString()
+      : throw WebScraperException(
+          'ERROR: Webpage need to be loaded first, try calling loadWebPage');
+
+  /// Returns List of elements titles found at specified address.
+  /// Example address: "div.item > a.title" where item and title are class names of div and a tag respectively.
+  /// For ease of access, when using Chrome inspection tool, right click the item you want to copy, then click "Inspect" and at the console, right click the highlighted item, right click and then click "Copy > Copy selector" and provide as String address parameter to this method.
+  List<String> getElementTitle(String address) {
+    if (_response == null) {

```

Comment:
Since you are creating 2 new methods for the package. I would suggest the following things:

- Add usage to README.md file
- Create simple unit tests in this [file](https://github.com/tusharojha/web_scraper/blob/master/test/web_scraper_test.dart).

Ps. please verify that the comment at line 135 works.

Reply:
I took a look at the unit testing and couldn't figure out how to do it. I can probably change the README fairly easily though.
Gonna take a look at that now.